packages/agreement-process/src/services/agreementService.ts

```diff
@@ -986,7 +987,8 @@ export function agreementServiceBuilder(
         authData,
         correlationId,
         logger,
-      }: WithLogger<AppContext<UIAuthData | M2MAdminAuthData>>
+      }: WithLogger<AppContext<UIAuthData | M2MAdminAuthData>>,
+      delegationId?: DelegationId
```

Comment:
This will make the function call less error prone since it constricts to acknowledge `delegationId` parameter.
Plus, can you put it before the context?
```suggestion
      delegationId: DelegationId | undefined
```